File: spec/javascript/test_enum_to_i.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('EnumToI', 'src/enum_0.bin', function(r, EnumToI) {
  assert.equal(r.pet1, EnumToI.Animal.CAT);
  assert.equal(r.pet2, EnumToI.Animal.CHICKEN);

  assert.equal(r.pet1I, 7);
  assert.equal(r.oneLtTwo, true);
});

File: spec/python/test_enum_to_i.py
import unittest

from enum_to_i import EnumToI

class TestEnumToI(unittest.TestCase):
    def test_enum_to_i(self):
        r = EnumToI.from_file("src/enum_0.bin")

        self.assertEqual(r.pet_1, EnumToI.Animal.cat)
        self.assertEqual(r.pet_2, EnumToI.Animal.chicken)

        self.assertEqual(r.pet_1_i, 7)
        self.assertEqual(r.one_lt_two, True)
